madminer/utils/morphing.py

```diff
@@ -573,16 +988,30 @@ def _propose_basis(self, fixed_benchmarks, n_missing_benchmarks):
         return basis
 
     def _draw_random_thetas(self, n_thetas):
-        """ Randomly draws parameter vectors within the specified parameter ranges. """
+        """Randomly draws parameter vectors within the specified parameter ranges."""
 
         # First draw random numbers in range [0, 1)^n_parameters
         u = np.random.rand(n_thetas, self.n_parameters)
 
         # Transform to right range
-        thetas = self.parameter_range[:, 0] + u * (self.parameter_range[:, 1] - self.parameter_range[:, 0])
+        thetas = self.parameter_range[:, 0] + u * (
+            self.parameter_range[:, 1] - self.parameter_range[:, 0]
+        )
 
         return thetas
 
+    def _calculate_component_weight(self, theta):
+        """Calculate the component weights for the given theta"""
+        component_weights = np.zeros(self.n_components)
+        for c in range(self.n_components):
+            factor = 1.0
+            for p in range(self.n_parameters):
+                factor *= float(theta[p] ** self.components[c, p])
+            component_weights[c] = factor
+        component_weights = np.array(component_weights)
+
+        return component_weights
```

Comment:
No need to separate `np.array` conversion, and `return` clause in 2 different lines.

```suggestion
        return np.array(component_weights)
```

Reply:
@wangzhe04 remember to delete [this line](https://github.com/wangzhe04/madminer/blob/d3caf7971155cda1cb7a23330e86ecce4ac4fd99/madminer/utils/morphing.py#L1009) above the one I suggested, as it is now useless.

Reply:
Thanks for reminding me!, I have deleted in the new commit.